src/embeddings.py

```
@@ -0,0 +1,224 @@
+
+
+class BatchItem(NamedTuple):
+    """A single item in a batch with its metadata."""
+
+    text: str
+    text_id: str
+    chunk_index: int
+
+
+class EmbeddingClient:
+    def __init__(self, api_key: str | None = None):
```

Ruff (0.11.9)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)